student-store-ui/src/components/App/App.jsx

```diff
@@ -5,14 +5,47 @@ import Home from "../Home/Home";
 import "./App.css";
 import { BrowserRouter } from "react-router-dom";
 import axios from "axios";
-import defaultCart from "../../constants/cart";
 
 const API_LINK = "https://codepath-store-api.herokuapp.com/store";
 
 export default function App() {
-  const [cart, setCart] = React.useState(defaultCart);
+  const [cart, setCart] = React.useState({});
   const [products, setProducts] = React.useState();
-  const updateProductInCart = (id, count) => setCart((prev) => ({ ...prev, [id]: count }));
+
+  const updateProductInCart = (id, count) => {
+    setCart((prev) => {
+      // don't update if products haven't loaded
+      if (!products) return prev;
```

Comment:
Even for 1 liners, we should still make it 

if (!products) {
return prev;
}